GMX-Bots/Large-Open-and-Increase-Positions/src/utils.ts

```diff
@@ -0,0 +1,50 @@
+import BigNumber from "bignumber.js";
+import { Finding, FindingSeverity, ethers, FindingType } from "forta-agent";
+
+export function ethersBnToBn(value: ethers.BigNumber, decimals: number): BigNumber {
+  return new BigNumber(value.toString()).shiftedBy(-decimals);
+}
+
+export const createFinding = (
+  account: string,
+  vaultAddress: string,
+  updatePositionKey: string,
+  increasePositionKey: string,
+  positionSize: ethers.BigNumber,
+  positionSizeDelta: ethers.BigNumber,
+  positionSizeDifference: ethers.BigNumber
+): Finding => {
+  if (updatePositionKey === increasePositionKey && ethers.BigNumber.from(positionSize).eq(positionSizeDelta)) {
+    return Finding.fromObject({
+      name: "Large position size opened on GMX's Vault Contract",
+      description: "UpdatePosition event emitted with a large position size",
+      alertId: "GMX-1-1",
+      severity: FindingSeverity.Info,
+      type: FindingType.Info,
+      protocol: "GMX",
+      metadata: {
+        gmxVault: vaultAddress,
+        account: account,
+        positionSize: ethersBnToBn(positionSizeDelta, 30).decimalPlaces(2).toString(),
+        positionKey: updatePositionKey,
+      },
+    });
+  } else {
```

Comment:
what if `updatePositionKey != increasePositionKey && positionSize.eq(positionSizeDelta)` ? do we still need to create this finding? or it should be else if `updatePositionKey === increasePositionKey && !positionSize.eq(positionSizeDelta)`

Reply:
added a logic to check that the keys of both `increasePosition` and `updatePosition` events match. Thanks for pointing that out :)